source/common/common/regex.h

```diff
@@ -66,7 +90,28 @@ class Utility {
   static CompiledMatcherPtr parseRegex(const RegexMatcherType& matcher) {
     // Google Re is the only currently supported engine.
     ASSERT(matcher.has_google_re2());
-    return std::make_unique<CompiledGoogleReMatcher>(matcher);
+    return std::make_unique<CompiledGoogleReMatcher>(matcher.regex(), matcher.google_re2());
+  }
+
+  /**
+   * Construct a compiled regex matcher from a match config.
+   */
+  template <class RegexMatcherType = xds::type::matcher::v3::RegexMatcher>
+  static CompiledMatcherPtr parseRegex(const RegexMatcherType& matcher,
+                                       ProtobufMessage::ValidationVisitor& validation_visitor) {
+    // Fall back to deprecated google_re2 field.
+    if (matcher.has_google_re2()) {
+      return parseRegex(matcher);
+    }
+
+    auto* factory = Config::RegistryUtils::getFactory<CompiledMatcherFactory>(matcher.engine());
```

Comment:
To build a compiled regex matcher from a typed config, this method requires `Config::Utility::getFactory`. However, this file cannot include `utility.h`, or there will be a circular reference and cause a compilation error in Bazel.

So I decoupled and moved these methods from `utility.h` into a new file `registry_utils.h`. Is this acceptable?

Reply:
I would like to ask people who may be familiar with these kind of build problems. Do you have some opinions on it? @lizan @phlax